rpc/src/v1/impls/personal.rs

```diff
@@ -104,37 +133,21 @@ impl<D: Dispatcher + 'static> Personal for PersonalClient<D> {
 		}
 	}
 
-	fn send_transaction(&self, meta: Metadata, request: TransactionRequest, password: String) -> BoxFuture<RpcH256, Error> {
-		let dispatcher = self.dispatcher.clone();
-		let accounts = try_bf!(self.account_provider());
-
-		let default = match request.from.as_ref() {
-			Some(account) => Ok(account.clone().into()),
-			None => accounts
-				.dapp_default_address(meta.dapp_id().into())
-				.map_err(|e| errors::account("Cannot find default account.", e)),
-		};
-
-		let default = match default {
-			Ok(default) => default,
-			Err(e) => return Box::new(future::err(e)),
-		};
+	fn sign_transaction(&self, meta: Metadata, request: TransactionRequest, password: String) -> BoxFuture<RpcH256, Error> {
```

Comment:
You need to get whole `RichRawTransaction`, just the hash is not useful at all. The whole point of this RPC method is to get signed RLP (raw transaction) that you will be able to propagate later.